### ISSUE TITLE
Incorrect list of sources in /reverse documentation

### ISSUE THREAD

**Issue by meelkor** [pelias/pelias-doc#201]
I am not sure whether is an issue of the documentation or the service itself, but:

As of now, when trying to request reverse geocoding using geonames source, I get following message:

`/reverse does not support geonames`

even though documentation specifies `geonames` as one of the supported sources. 

Whole response:
```
{
    "geocoding": {
        "version": "0.2",
        "attribution": "https://search.mapzen.com/v1/attribution",
        "query": {
            "layers": [
                "locality",
                "county",
                "region",
                "country"
            ],
            "sources": [
                "geonames"
            ],
            "size": 5,
            "private": false,
            "point.lat": 19.062198992675675,
            "point.lon": 72.86285161972046,
            "boundary.circle.radius": 1,
            "boundary.circle.lat": 19.062198992675675,
            "boundary.circle.lon": 72.86285161972046,
            "lang": {
                "name": "English",
                "iso6391": "en",
                "iso6393": "eng",
                "defaulted": false
            },
            "querySize": 20
        },
        "errors": [
            "/reverse does not support geonames"
        ],
        "engine": {
            "name": "Pelias",
            "author": "Mapzen",
            "version": "1.0"
        },
        "timestamp": 1498680013867
    },
    "type": "FeatureCollection",
    "features": []
}
```

**Comment by orangejulius** [comment]
Hey @meelkor,
This is a bug in the error message being returned. It should read "coarse reverse does not support geonames". Geonames only has point data for admin areas like locality, region, country, etc, so it's not good at answering the question "what admin areas is this point inside of?", which is essentially what reverse geocoding for admin layers means. In the past, we implemented this functionality by finding the nearest points anyways, but that approach fails more often than not. For example, if you asked for the country a few miles outside The Vatican, Pelias using Geonames data would have given The Vatican as the answer, since the point for The Vatican was closest. It wouldn't know that the area was actually part of Italy.

In order to do real and correct coarse reverse geocoding, polygon information is required.  The only data source Pelias currently supports with this information is [Who's on First](http://whosonfirst.mapzen.com/), so once we implemented correct coarse reverse geocoding, we've effectively dropped support for all other sources.

Regular reverse queries (such as finding which venue or address is at a point) will continue to support geonames. For your query in particular I'd recommend not specifying sources at all. It will do the right thing.

Let us know if you have any other questions. We'll also update the error message and documentation to be more clear.
